controller/engine_controller.go

```diff
@@ -951,6 +961,11 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 				return
 			}
 			engine = e
+			if existingEngine.Status.LastRestoredBackup != engine.Status.LastRestoredBackup {
+				if err := m.acquireRestoringCounter(false); err != nil {
+					m.logger.WithError(err).Error("Failed to unacquire restoring counter")
+				}
+			}
```

Comment:
As discussed, need to take care of the non-successful restore to decrease the concurrent count to avoid starvation.

Reply:
Updated.